dartagnan/src/main/java/com/dat3m/dartagnan/expression/LastValueInterface.java

```diff
@@ -0,0 +1,10 @@
+package com.dat3m.dartagnan.expression;
+
+import org.sosy_lab.java_smt.api.Formula;
+import org.sosy_lab.java_smt.api.SolverContext;
+
+public interface LastValueInterface extends ExprInterface {
```

Comment:
It will not work but not due to this interface, but because of the parser. The grammar only accepts variables (`Location` or `Register`) and constants. Thus this interface is in sync with the parser.
`herd7` does not accept either the test you proposed so I don't really see the point in having support for this.

Reply:
In general it seems like an artificial restriction though, but it seems unlikely that we ever need it. So I'm fine with this.